### PR TITLE
Theme pragma SUPPORT_THEME_HOOKS

### PR DESCRIPTION
**Pull request by gerhardol**
Follow up to https://github.com/gitextensions/gitextensions/pull/11006
Credit to @SimonCropp , fork was removed though

## Proposed changes

Enclose unused classes and methods related to theming with the pragma SUPPORT_THEME_HOOKS

## Test methodology 

Build

## Merge strategy

I agree that the maintainer squash merge this PR (if the commit message is clear).

----

:black_nib: I contribute this code under [The Developer Certificate of Origin](../blob/master/contributors.txt).

### CODE REVIEW

File: GitUI/Theming/Renderers/ButtonRenderer.cs [hunk 1]
```diff
@@ -281,7 +281,8 @@ public enum CheckBox
                 CBS_EXCLUDEDDISABLED = 20,
             }
 
-            public enum GroupBox
+ #if SUPPORT_THEME_HOOKS
+           public enum GroupBox
```
Comment (mstv) [inline GitUI/Theming/Renderers/ButtonRenderer.cs:285]:
```suggestion
            public enum GroupBox
```

File: GitUI/Theming/Renderers/EditRenderer.cs [hunk 2]
```diff
@@ -185,13 +185,15 @@ public enum Background
                 EBS_ASSIST = 6,
             }
 
-            public enum BackgroundWithBorder
+ #if SUPPORT_THEME_HOOKS
+           public enum BackgroundWithBorder
```
Comment (mstv) [inline GitUI/Theming/Renderers/EditRenderer.cs:189]:
```suggestion
            public enum BackgroundWithBorder
```

File: GitUI/Theming/Renderers/SpinRenderer.cs [hunk 3]
```diff
@@ -139,7 +139,8 @@ public enum Down
                 DNS_DISABLED = 4,
             }
 
-            public enum UpHorizontal
+ #if SUPPORT_THEME_HOOKS
+           public enum UpHorizontal
```
Comment (mstv) [inline GitUI/Theming/Renderers/SpinRenderer.cs:143]:
```suggestion
            public enum UpHorizontal
```

**Review by mstv — COMMENTED** [review]
There is no need to leave nits for me. :)

**Review by mstv — APPROVED** [review]

File: GitUI/Theming/Renderers/ComboBoxRenderer.cs [hunk 4]
```diff
@@ -268,13 +270,15 @@ public enum Readonly
                 CBRO_DISABLED = 4,
             }
 
-            public enum CueBanner
+ #if SUPPORT_THEME_HOOKS
+           public enum CueBanner
```
Comment (RussKie) [inline GitUI/Theming/Renderers/ComboBoxRenderer.cs:274]: Another nit :)

**Review by RussKie — APPROVED** [review]

**Pull request merged**